lib/package.gi

```diff
@@ -226,35 +294,50 @@ BindGlobal( "AddPackageInfos", function( files, pkgdir, ignore )
         version:= record.Version;
 
         # If we have this version already then leave it out.
-        if ForAll( GAPInfo.PackagesInfo,
-                    r ->    r.PackageName <> record.PackageName
-                        or r.Version <> version ) then
-
-          # Check whether GAP wants to reset loadability.
-          if     IsBound( GAPInfo.PackagesRestrictions.( pkgname ) )
-              and GAPInfo.PackagesRestrictions.( pkgname ).OnInitialization(
-                      record ) = false then
-            Add( GAPInfo.PackagesInfoRefuseLoad, record );
-          elif pkgname in ignore then
-            LogPackageLoadingMessage( PACKAGE_DEBUG,
-                Concatenation( "ignore package ", record.PackageName,
-                " (user preference PackagesToIgnore)" ), "GAP" );
-          else
-            record.InstallationPath:= Filename( [ pkgdir ], file[2] );
-            if not IsBound( record.PackageDoc ) then
-              record.PackageDoc:= [];
-            elif IsRecord( record.PackageDoc ) then
-              record.PackageDoc:= [ record.PackageDoc ];
-            fi;
-            if IsHPCGAP then
-              # FIXME: we make the package info record immutable, to
-              # allow access from multiple threads; but that in turn
-              # can break packages, which rely on their package info
-              # record being readable (see issue #2568)
-              MakeImmutable(record);
-            fi;
-            Add( GAPInfo.PackagesInfo, record );
+        if ForAny( GAPInfo.PackagesInfo,
+                    r -> r.PackageName = record.PackageName
+                    and r.Version = version ) then
+          continue;
+        fi;
+
+        # Check whether GAP wants to reset loadability.
+        if     IsBound( GAPInfo.PackagesRestrictions.( pkgname ) )
+            and GAPInfo.PackagesRestrictions.( pkgname ).OnInitialization(
+                    record ) = false then
+          Add( GAPInfo.PackagesInfoRefuseLoad, record );
+        elif pkgname in ignore then
+          LogPackageLoadingMessage( PACKAGE_DEBUG,
+              Concatenation( "ignore package ", record.PackageName,
+              " (user preference PackagesToIgnore)" ), "GAP" );
+        else
+          record.InstallationPath:= Filename( [ pkgdir ], file[2] );
+          if not IsBound( record.PackageDoc ) then
+            record.PackageDoc:= [];
+          elif IsRecord( record.PackageDoc ) then
+            record.PackageDoc:= [ record.PackageDoc ];
+          fi;
+
+          if IsBound( record.HasSubpackage ) and record.HasSubpackage = true then
+            dir := record.InstallationPath;
+            for name in SubdirectoriesOfPath( dir ) do
+              files := FindPackageInfosInSubdirectories( name );
+              AddPackageInfos( files, Directory( dir ), ignore, record );
+            od;
           fi;
+
+          if isSubpackage then
```

Comment:
```suggestion
          if isSubPackage then
```

But actually, I think you should change it the other way around, and use `isSubpackage` everywhere instead of `isSubPackage`.

Reply:
I pushed a commit to resolve this.